sentinel/test/mocha/lib/message-utils.js

```diff
@@ -0,0 +1,331 @@
+        .should.equal('foo');
+    });
+    describe('recipient variations', () => {
+      const fromPhone = 'fromPhone',
+            clinicPhone = 'clinicPhone',
+            parentPhone = 'parentPhone',
+            grandparentPhone = 'grandParentPhone',
+            fieldsPhone = 'fieldsPhone',
+            inlinePhone = 'inlinePhone',
+            complexInlinePhone = 'complexInlinePhone';
+      const doc = {
+        form: 'x',
+        from: fromPhone,
+        fields: {
+          phone: fieldsPhone,
+        },
+        phone: inlinePhone,
+        contact: {
+          parent: {
+            type: 'clinic',
+            contact: {
+              phone: clinicPhone
+            },
+            parent: {
+              type: 'health_center',
+              contact: {
+                phone: parentPhone
+              },
+              parent: {
+                type: 'district_hospital',
+                contact: {
+                  phone: grandparentPhone
+                }
+              }
+            }
+          }
+        },
+        complex: { inline: { phone: complexInlinePhone }}
+      };
+
+      it('resolves reporting_unit correctly', () => {
+        utils._getRecipient(doc, 'reporting_unit')
+          .should.equal(fromPhone);
+      });
+      it('resolves clinic correctly', () => {
+        utils._getRecipient(doc, 'clinic')
+          .should.equal(clinicPhone);
+      });
+      it('resolves parent correctly', () => {
+        utils._getRecipient(doc, 'parent')
+          .should.equal(parentPhone);
+      });
+      it('resolves grandparent correctly', () => {
+        utils._getRecipient(doc, 'grandparent')
+          .should.equal(grandparentPhone);
+      });
+    });
+    it('tries to resolve the value from the fields property', () => {
+      utils._getRecipient({fields: {foo: 'bar'}}, 'foo')
+        .should.equal('bar');
+    });
+    it('tries to resolve simple values directly on the doc', () => {
+      utils._getRecipient({foo: 'bar'}, 'foo')
+        .should.equal('bar');
+    });
+    it('tries to resolve complex values directly on the doc', () => {
+      utils._getRecipient({foo: {bar: {smang: 'baz'}}}, 'foo.bar.smang')
+        .should.equal('baz');
+    });
+    it('is doc.from if the recipient cannot be resolved', () => {
+      utils._getRecipient({from: 'foo'}, 'a-recipient')
+        .should.equal('foo');
+    });
+  });
+
+  describe('generate', () => {
+
+    it('adds uuid', done => {
+      sinon.stub(uuid, 'v4').returns('some-uuid');
+      const config = {};
+      const translate = null;
+      const doc = {};
+      const content = { message: 'xxx' };
+      const recipient = '+1234';
+      const messages = utils.generate(config, translate, doc, content, recipient);
+      expect(messages.length).to.equal(1);
+      const message = messages[0];
+      expect(message.message).to.equal('xxx');
+      expect(message.to).to.equal('+1234');
+      expect(message.uuid).to.equal('some-uuid');
```

Comment:
Could actually test if it sets a valid UUID here instead of stubbing.

Reply:
I actually think testing the stubbing is more useful. I don't feel the need to test that the `uuid` library generates a valid uuid - I _hope_ they have sufficient testing to ensure that. This test ensures that we call the `uuid` library and use the result.